flake.nix

```
@@ -7,6 +7,10 @@
       path = builtins.path { path = ./example; filter = path: _: baseNameOf path == "flake.nix"; };
     };
 
+    # NOTE: this is a lib factory, as it takes: { pkgs, lib }
+    # so before using we need to pass it to pkgs.callPackage
+    lib = import ./nix/lib.nix;
```

```suggestion
    lib = {
      withPkgs = import ./nix/lib.nix;
    };
```

This leaves room for functions that don't depend on `pkgs`.
Accepting a `lib` attribute in the parameter is implied.